crates/iceberg/src/scan.rs

```diff
@@ -55,17 +55,23 @@ pub struct TableScanBuilder<'a> {
     batch_size: Option<usize>,
     case_sensitive: bool,
     filter: Option<Predicate>,
+    concurrency_limit_manifest_files: usize,
+    concurrency_limit_manifest_entries: usize,
 }
 
 impl<'a> TableScanBuilder<'a> {
     pub(crate) fn new(table: &'a Table) -> Self {
+        let num_cpus = num_cpus::get();
```

Comment:
Not a reivew comment, just a glimpse of this PR, we can replace this with `std::thread::available_parallelism()`, `num_cpus` is almost unmaintained (as it no longer needs to), and has issues that are solved by the std counterpart, e.g., this one: [std::thread::available_parallelism() return 0](https://github.com/rust-lang/rust/issues/115868).